File: Leetcode/Daily/12-06-2024/prob.cpp
class Solution {
public:
    void sortColors(vector<int>& v) {
        int n = v.size();
        int l = 0,mid = 0,h = n-1;

        while(mid<=h){
            if(v[mid]==0){
                swap(v[l],v[mid]);
                l++;
                mid++;
            }
            else if(v[mid]==1)mid++;
            else{
                swap(v[mid],v[h]);
                h--;
            }
        }
    }
};

File: Leetcode/Daily/12-06-2024/prob.go
func swap(a, b *int){
    temp := *b
    *b = *a
    *a = temp
}
func sortColors(v []int)  {
    l, mid, h := 0,0, len(v)-1

    for mid<=h{
        if v[mid]==0{
            v[mid],v[l] = v[l],v[mid]
            l++
            mid++
        }else if v[mid]==1{
            mid++
        }else{
            v[mid],v[h] = v[h],v[mid]
            h--
        }
    }
}